server/src/main/java/org/apache/druid/catalog/model/table/InputFormats.java

```diff
@@ -262,13 +265,37 @@ public static class GenericFormatDefn extends BaseFormatDefn
     public static final String INPUT_FORMAT_SPEC_PROPERTY = "inputFormatSpec";
     public static final String FORMAT_KEY = "generic";
 
+    private static class FormatPropertyDefn extends SimplePropertyDefn<InputFormat>
+    {
+      public FormatPropertyDefn()
+      {
+        super(
+            INPUT_FORMAT_SPEC_PROPERTY,
+            InputFormat.class,
+            PropertyAttributes.merge(
+                ImmutableMap.of(
+                    PropertyAttributes.SQL_JAVA_TYPE,
+                    String.class
+                ),
+                PropertyAttributes.OPTIONAL_SQL_FN_PARAM
```

Comment:
Why is this property optional, I thought it would be required; seems that it defines this format?